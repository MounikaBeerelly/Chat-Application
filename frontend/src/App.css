h1 {
  color: red;
}
button {
  text-align: center;
}

li {
  font-size: 20px;
}
