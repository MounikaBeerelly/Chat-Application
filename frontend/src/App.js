import React, { Component } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import './App.css';

import RegistrationComponent from './components/registartionPage';
import LoginComponent from './components/loginPage';
import ProfilePageComponent from './components/profile/profilePage';
import Homecomponent from './components/homePage';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="container">
          <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <div class="navbar-header">
              <Link to={'/'} className="navbar-brand">Chat Application</Link>
            </div>
            <div className="collapse navbar-collapse" id="navbarSupportedContent">
              <ul className="navbar-nav mr-auto">
                <li className="nav-item">
                  <Link to={'/'} className="nav-link">Home</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/profile'} className="nav-link">Profile</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/register'} className="nav-link">Register</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/login'} className="nav-link">Login</Link>
                </li>
              </ul>
            </div>
          </nav>
          <Switch>
            <Route exact path='/' component={Homecomponent} />
            <Route path='/profile' component={ProfilePageComponent} />
            <Route path='/register' component={RegistrationComponent} />
            <Route path='/login' component={LoginComponent} />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;

