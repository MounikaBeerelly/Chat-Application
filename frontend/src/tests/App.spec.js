import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
describe('sample test cases', () => {

    test('should equal', () => {
        const component = renderer.create(<App />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})