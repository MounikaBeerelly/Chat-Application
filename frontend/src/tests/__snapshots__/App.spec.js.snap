// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sample test cases should equal 1`] = `
<div
  className="container"
>
  <div
    className="jumbotron"
  >
    <h3>
      Chat Application
    </h3>
  </div>
  <div
    className="container"
  >
    <h2
      style={
        Object {
          "color": "blue",
          "fontStyle": "italic",
          "textAlign": "center",
        }
      }
    >
      Login Form
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <input
          className="form-control"
          id="username"
          required={true}
          type="text"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="btn-primary"
        type="submit"
      >
        Login
      </button>
    </form>
    <br />
     
    <br />
    <h5>
      Don't have an account? 
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
       
    </h5>
  </div>
</div>
`;
